sshuttle/client.py

```diff
@@ -1038,12 +1011,13 @@ def main(listenip_v6, listenip_v4,
              redirectport_v6, redirectport_v4, dnsport_v6, dnsport_v4,
              required.udp)
 
-    # start the watchdog for acl files
-    if (acl_file):
-        start_acl_watchdog(acl_file, ALLOWED_ACL_TYPE)
-        start_acl_watchdog(disallowed_acl_file, DISALLOWED_ACL_TYPE)
-        start_acl_watchdog(acl_sources_file, ACL_SOURCES_TYPE)
-        start_acl_watchdog(acl_excluded_sources_file, ACL_EXCLUDED_SOURCES_TYPE)
+    if (REDIS_HOST is None or REDIS_PORT is None):
+        raise Fatal("REDIS_HOST and REDIS_PORT environment variables must both be set!")
+
+    channelSubscriptions = [sshuttleAclEventsChannel]
+    channelListener = ChannelListener(REDIS_HOST, REDIS_PORT, channelSubscriptions)
```

Comment:
I don't see any logic about disconnect handler...is this all handled by redis library that you are using?

Reply:
Investigating this now. There does seem to be an issue here. Just working out the details.